### PR TITLE
Error instead of panic on timestamp overflow

### PR DESCRIPTION
**Pull request by progval**
I would have prefered either returning a null value or trying to encode as a non-nanosecond timestamp (sometimes possible if the ORC data does not use nanosecond-level precision), but this would require more significant changes to the code.

### CODE REVIEW

File: tests/basic/data/pyarrow_timestamps.orc
Comment (waynexia) [inline tests/basic/data/pyarrow_timestamps.orc:1]: Looks like this file is not changed?

**Review by waynexia — APPROVED** [review]
Looks good, thanks @progval 

I'm curious why `pyarrow_timestamps.orc` is present in the diff file list.

**Comment by waynexia** [comment]
Great 👍 I'm going to merge this

**Comment by Jefffrey** [comment]
Thanks for this :+1:

**Pull request merged**